src/cpp/shared/combat/animation.cpp

```diff
@@ -1,5 +1,152 @@
+#include <string.h>
 #include "combat/animation.h"
 
+void BuildTeleporterTempWalkSeq(SMonFrameInfo *frm, int last, int notFirst, bool closeMove) {
```

Comment:
It's nice that these are in a separate file to make them easier to follow. As a first candidate for refactoring, I think all of this can be done declaratively -- you can write down some data that dictates how these animations will be created, and it should be easier to read.